### PR TITLE
task(FXA-10804): Fast-follow apply upgrade process to signin token code

### PR DESCRIPTION
**Pull request by dschom**
## Because
- We already landed a PR for 10804 that does this for totp and unblock codes. In that review, another place where this could be done was spotted.
- We didn't want to block a release, so this is being done as a fast follow.

## This pull request

- Adds the ability to upgrade key stretching after providing a signin token code
- Adds functional tests to validate the upgrade.

## Issue that this pull request solves

Closes: 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      await expect(page).toHaveURL(/settings/);
+
+      // Make sure upgrade occurred
+      if (signinVersion.version === 2) {
+        const client = target.createAuthClient(2);
+        const status = await client.getCredentialStatusV2(email);
+        expect(status.currentVersion).toEqual(`v2`);
+      }
+
+      // Delete blocked account, required before teardown
```
Comment (vpomerleau) [inline packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts:58]: Is the account blocked?

File: packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ */
+test.describe('severity-2 #smoke', () => {
+  type Version = { version: 1 | 2; query: string };
+  type TestCase = { signinVersion: Version };
+  const v1: Version = { version: 1, query: '' };
+  const v2: Version = { version: 2, query: 'stretch=2' };
+  const TestCases: TestCase[] = [{ signinVersion: v1 }, { signinVersion: v2 }];
+
+  for (const { signinVersion } of TestCases) {
+    test(`fffff signs up as v1, sign in within token code, signs  as v${signinVersion.version}`, async ({
```
Comment (vpomerleau) [inline packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts:22]: fffff?

File: packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        signin,
+        connectAnotherDevice,
+        deleteAccount,
+        signinTokenCode,
+      },
+      testAccountTracker,
+    }, {}) => {
+      const { email, password } = await testAccountTracker.signUpSync();
+      await page.goto(
+        `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email&${signinVersion.query}`
```
Comment (vpomerleau) [inline packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts:36]: Do we want to continue using fx_desktop_v3 for tests, or switch to oauth?

File: packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx [hunk 4]
```diff
@@ -102,17 +103,20 @@ function resetMockSensitiveDataClient() {
...
+async function render(mocks: Array<MockedResponse>) {
   renderWithLocalizationProvider(
-    <LocationProvider>
-      <SigninTokenCodeContainer
-        {...{
-          integration,
-        }}
-      />
-    </LocationProvider>
+    <MockedProvider mocks={mocks} addTypename={false}>
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx:111]: Is this change necessary if we don't pass in mocks in any of the render calls?

**Review by vpomerleau — APPROVED** [review]
LGTM, just a few questions!

File: packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts (hunk 1 above)
Comment (dschom) [reply packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts:58]: Nope. Just some copy and pasta. I'll fix that.

File: packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx (hunk 4 above)
Comment (dschom) [reply packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx:111]: It appears to be. I was getting an error without it. When the control renders, the 'useMutation' functions get called, and that errors if there's MockedProvider wrapping the component.

File: packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts (hunk 3 above)
Comment (dschom) [reply packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts:36]: Ah good question. There's a lot of uses of fx_desktop_v3 in our functional tests. I think cleaning this up would be a good followup.

File: packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx (hunk 4 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx:111]: In that case I'm surprised it wasn't needed for signin_totp_code

Comment (dschom) [reply packages/fxa-settings/src/pages/Signin/SigninTokenCode/container.test.tsx:111]: Just took a second look at this, it's because of a mock in `SignInTokenCode` on  `jest.spyOn(ApolloModule, 'useMutation')`. I feel like the more consistent way to do this is using the `MockedProvider`.
